### ISSUE TITLE
The Sdk Has Not Been Initialized, Make sure to call FacebookSdk.sdkInitialize() first

### ISSUE THREAD

**Issue by architect-aimonkey** [thebergamo/react-native-fbsdk-next#241]
## 🐛 Bug Report

When upgrading react-native-fbsdk-next from v7.3.3 to v8+ our iOS and Android app crashes on start
The Metro server reports: 
"The Sdk Has Not Been Initialized, Make sure to call FacebookSdk.sdkInitialize() first "
I read that the method FacebookSdk.sdkInitialize() is deprecated and tried all the recommendations posted elsewhere but no success. 

## To Reproduce

Upgrade react-native-fbsdk-next from v7.3.3 to v8+

## Expected Behavior

App runs normally

## Environment

System:
    OS: Windows 10 10.0.22000
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 9.45 GB / 39.81 GB
  Binaries:
    Node: 14.19.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.23.0-20200615.1913 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK:
      System Images: android-25 | Google APIs Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom_64, android-30 | Google Play Intel x86 Atom_64, android-31 | Google Play Intel x86 Atom_64       
      Android NDK: Not Found    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
  IDEs:
    Android Studio: Not Found
    Visual Studio: 16.10.31424.327 (Visual Studio Community 2019)
  Languages:
    Java: 15.0.1
    Python: 3.10.2
  npmPackages:
    @react-native-community/cli: Not Found
    react: ^17.0.2 => 17.0.2
    react-native: ^0.67.4 => 0.67.4
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

**Comment by muja555** [comment]
I solved it by adding `facebookSdkVersion = "13.1.0"` in my `myproject/android/build.gradle` 
```
buildscript {
    ext {
       ...
       facebookSdkVersion = "13.1.0"
  }
 }

**Comment by architect-aimonkey** [comment]
thanks! but that doesn't work for me, now I'm getting
the following error:

D:\AIMONKEY\Develop\node_modules\react-native-fbsdk-next\android\src\main\java\com\facebook\reactnative\androidsdk\Utility.java:172: error: cannot find symbol
        contentBuilder.setImageUrl(url != null ? Uri.parse(url) : null);
                      ^
  symbol:   method setImageUrl(url != nul[...] null)
  location: variable contentBuilder of type Builder
D:\AIMONKEY\Develop\node_modules\react-native-fbsdk-next\android\src\main\java\com\facebook\reactnative\androidsdk\Utility.java:173: error: cannot find symbol
        contentBuilder.setContentDescription(getValueOrNull(shareLinkContent, "contentDescription"));
                      ^
  symbol:   method setContentDescription(String)
  location: variable contentBuilder of type Builder
D:\AIMONKEY\Develop\node_modules\react-native-fbsdk-next\android\src\main\java\com\facebook\reactnative\androidsdk\Utility.java:174: error: cannot find symbol
        contentBuilder.setContentTitle(getValueOrNull(shareLinkContent, "contentTitle"));
                      ^
  symbol:   method setContentTitle(String)
  location: variable contentBuilder of type Builder
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\AIMONKEY\Develop\node_modules\react-native-fbsdk-next\android\src\main\java\com\facebook\reactnative\androidsdk\Utility.java uses unchecked or unsafe operations.

**Comment by mikehardy** [comment]
Have you investigated the logs? This looks like a duplicate of #236 / failure to read release notes / did not add client access token. Can you confirm you have added a client access token as mentioned in release notes? Can you watch Xcode console or adb logcat to find the actual error messages?

**Comment by Md-Mudassir47** [comment]
> I solved it by adding `facebookSdkVersion = "13.1.0"` in my `myproject/android/build.gradle`
> 
> ```
> buildscript {
>     ext {
>        ...
>        facebookSdkVersion = "13.1.0"
>   }
>  }
> ```

still breaking in android after adding the SDK version.

"react-native": "^0.67.4",
"react-native-fbsdk-next": "^8.0.1",

**Comment by mikehardy** [comment]
@Md-Mudassir47 

Why ignore my question? Not a great way to access support 🤔  "I'm having a problem, please help, but I won't take action on the help provided" ?

> Have you investigated the logs? This looks like a duplicate of https://github.com/thebergamo/react-native-fbsdk-next/issues/236 / failure to read release notes / did not add client access token. Can you confirm you have added a client access token as mentioned in release notes? Can you watch Xcode console or adb logcat to find the actual error messages?

**Comment by Md-Mudassir47** [comment]
> @Md-Mudassir47
> 
> Why ignore my question? Not a great way to access support 🤔 "I'm having a problem, please help, but I won't take action on the help provided" ?
> 
> > Have you investigated the logs? This looks like a duplicate of #236 / failure to read release notes / did not add client access token. Can you confirm you have added a client access token as mentioned in release notes? Can you watch Xcode console or adb logcat to find the actual error messages?

Apologies, Had not read the last message & thanks [this](https://github.com/thebergamo/react-native-fbsdk-next/issues/236#issuecomment-1111744717) helped me resolve the issue.

**Comment by mikehardy** [comment]
Great! Glad that worked. I'm going to close this as a duplicate of that one then since it seems like the same problem, and I'll pin that issue so it is more discoverable

**Comment by DiegoFawr** [comment]
> I solved it by adding `facebookSdkVersion = "13.1.0"` in my `myproject/android/build.gradle`
> 
> ```
> buildscript {
>     ext {
>        ...
>        facebookSdkVersion = "13.1.0"
>   }
>  }
> ```

Great!!
